### ISSUE TITLE
Deal with existing link id error from dlmgmtd

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#164]
The `OpteHdl::create_xde` method (in `opte-ioctl` create) can leave a dangling link id laying around; resulting in future requests to create an instance (which tries to create an xde device with the same name) to result in the following sled-agent error:

```
{"msg":"request completed","v":0,"name":"SledAgent","level":30,"time":"2022-06-24T04:08:02.851774675Z","hostname":"kalm","pid":27068,"uri":"/instances/0fb96d7c-0271-4a05-9957-7bad87cfed31","method":"PUT","req_id":"884ff07d-98a2-43a6-bd18-d4b090841aef","remote_addr":"[fd00:1122:3344:101::3]:41789","local_addr":"[fd00:1122:3344:101::1]:12345","component":"dropshot (SledAgent)","error_message_external":"Internal Server Error","error_message_internal":"Error managing instances: Instance error: Failure interacting with the OPTE ioctl(2) interface: netadm failed dlmgmtd: link id creation failed: 17","response_code":"500"}
```

The problem is that we do not delete the newly create link id in the case that the ioctl fails.

However, I also wonder if there are other ways we could get into this state (e.g. killing sled-agent after link id creation but before it runs the CreateXde ioctl) and if we should fine a way for opte-ioctl to detect this condition and cleanup the abandoned link id before continuing (or actually there might be an API for just searching for an existing link id, which would be perfect).

Basically, let's harden the xde device creation against situations like this.

**Comment by rzezeski** [comment]
While it would be nice to have logic that detects various mismatches of state between dlmgmtd/dls/opte, it doesn't seem like the thing to go deep on right now and also it might really be that the logic should live in sled-agent/omicron. Besides, it seems like sled-agent has changed how it creates xde devices since when this was first written, in that it seems to always increment the suffix anytime instance creation is retried or an instance is restarted. So for now I just made a small tweak to delete the link id if the creation of the xde device fails.

I tested this by unloading the xde device which causes instance creation to fail, and then verifying that the link is not present in dlgmtmd.

* sled-agent log

```
{"msg":"request completed","v":0,"name":"SledAgent","level":30,"time":"2022-07-28T01:14:14.084230506Z","hostname":"kalm","pid":664,"uri":"/instances/a754d7c2-c7ad-436f-9398-782824c447f5","method":"PUT","req_id":"106f0ab5-8390-4ade-9700-5b0b55c92506","remote_addr":"[fd00:1122:3344:101::3]:38342","local_addr":"[fd00:1122:3344:101::1]:12345","component":"dropshot (SledAgent)","error_message_external":"Internal Server Error","error_message_internal":"Error managing instances: Instance error: Failure interacting with the OPTE ioctl(2
) interface: command CreateXde failed: System { errno: 22, msg: \"underlay not initialized\" }","response_code":"500"}  
```

* dlmgmtd state

```
rpz@kalm:~$ pfexec mdb -p $(pgrep dlmgmtd) -e 'dlmgmt_id_avl::walk avl | ::printf "0x%p %s %d\n" dlmgmt_link_t . ll_link ll_linkid'
0x806d2a0 e1000g0 1
0x806d6e8 ixgbe0 2
0x806d760 ixgbe1 3
0x806d7d8 foo1 4
0x806d890 foo2 5
0x806d9c8 ryan0 6
0x806da40 net0 7
0x806dab8 net1 8
0x806dcd8 stub0 9
0x806dd50 underlay0 10
0x806ddc8 oxControlService0 11
0x806de40 oxControlStorage0 12
0x806deb8 oxControlStorage1 13
0x806df30 oxControlStorage2 14
0x806db30 oxControlStorage3 15
0x806dba8 oxControlStorage4 16
0x806dc20 oxControlService1 17
0x806c9d0 oxControlPublic0 18
0x806ca48 oxControlService2 19
```

**Comment by rzezeski** [comment]
Addressed in e48a59aae3e5bc08e47deb7b4f907de5cfa3f0f3.
